### PR TITLE
Fix : GemVersion TypeError unhashable type: 'list'

### PR DESCRIPTION
**Pull request by ziadhany**
Signed-off-by: ziadhany <ziadhany2016@gmail.com>

### CODE REVIEW

**Comment by TG1999** [comment]
@ziadhany please update your branch with latest main

**Comment by TG1999** [comment]
@ziadhany please add a test to verify your changes

File: tests/test_rubygems_gem_version.py [hunk 1]
```diff
@@ -181,9 +181,9 @@ def test_split_segments():
 
 
 def test_canonical_segments():
-    assert GemVersion("1.0.0").canonical_segments == [1]
-    assert GemVersion("1.0.0.a.1.0").canonical_segments == [1, "a", 1]
-    assert GemVersion("1.2.3-1").canonical_segments == [1, 2, 3, "pre", 1]
+    assert GemVersion("1.0.0").canonical_segments == (1,)
+    assert GemVersion("1.0.0.a.1.0").canonical_segments == (1, "a", 1)
+    assert GemVersion("1.2.3-1").canonical_segments == (1, 2, 3, "pre", 1)
```
Comment (pombredanne) [inline tests/test_rubygems_gem_version.py:186]:
```suggestion
    assert GemVersion("1.2.3-1").canonical_segments == (1, 2, 3, "pre", 1)

def test_GemVersion_is_hashable():
    hash(GemVersion("1.0.0"))
```

**Review by pombredanne — APPROVED** [review]
LGTM! we still would want to have a `hash()` test to validate we can hash a `GemVersion`... this is good to go otherwise.

**Comment by pombredanne** [comment]
@TG1999 if you want to add the test and merge, you can then cut a release IMHO :)

**Comment by pombredanne** [comment]
> @TG1999 if you want to add the test and merge, you can then cut a release IMHO :)

I committed the new test!

**Pull request merged**